src/include/common/csv_reader/csv_reader.h

```diff
@@ -30,11 +30,67 @@ struct CSVReaderConfig {
     bool hasHeader;
 };
 
-struct CSVDescription {
-    CSVDescription(const string filePath, const CSVReaderConfig csvReaderConfig)
-        : filePath{move(filePath)}, csvReaderConfig{move(csvReaderConfig)} {}
+struct CopyDescription {
+    CopyDescription(const string& filePath, const ReaderConfig readerConfig)
+        : filePath{filePath}, readerConfig{nullptr}, fileType{FileType::CSV} {
+        setFileType(filePath);
+        if (fileType == FileType::CSV) {
+            this->readerConfig = make_shared<ReaderConfig>(readerConfig);
+        }
+    }
+
+    enum class FileType { CSV, ARROW, PARQUET };
+
+    static string getFileTypeName(FileType fileType) {
+        switch (fileType) {
+        case FileType::CSV:
+            return "csv";
+
+        case FileType::ARROW:
+            return "arrow";
+
+        case FileType::PARQUET:
+            return "parquet";
+        }
+    }
+
+    static string getFileTypeSuffix(FileType fileType) { return "." + getFileTypeName(fileType); }
+
+    void setFileType(string const& fileName) {
+        auto csvSuffix = getFileTypeSuffix(FileType::CSV);
+        auto arrowSuffix = getFileTypeSuffix(FileType::ARROW);
+        auto parquetSuffix = getFileTypeSuffix(FileType::PARQUET);
+
+        if (fileName.length() >= csvSuffix.length()) {
+            if (!fileName.compare(
+                    fileName.length() - csvSuffix.length(), csvSuffix.length(), csvSuffix)) {
+                fileType = FileType::CSV;
+                return;
+            }
+        }
+
+        if (fileName.length() >= arrowSuffix.length()) {
+            if (!fileName.compare(
+                    fileName.length() - arrowSuffix.length(), arrowSuffix.length(), arrowSuffix)) {
+                fileType = FileType::ARROW;
+                return;
+            }
+        }
+
+        if (fileName.length() >= parquetSuffix.length()) {
+            if (!fileName.compare(fileName.length() - parquetSuffix.length(),
+                    parquetSuffix.length(), parquetSuffix)) {
+                fileType = FileType::PARQUET;
+                return;
+            }
+        }
+
+        throw CopyException("Unsupported file type: " + fileName);
+    }
+
     const string filePath;
-    const CSVReaderConfig csvReaderConfig;
+    shared_ptr<ReaderConfig> readerConfig;
```

Comment:
This does not seem to be a bug. There are a few places in the code base that copies the CopyDescription object, which leads to copying the unique_ptr. Since unique_ptr does not allow having multiple pointers pointing to the same object, it does not have a copy constructor. Therefore, an error about calling an implicitly-deleted copy constructor is triggered. Using shared_ptr avoids this issue.